codersdk/audit.go

```diff
@@ -0,0 +1,53 @@
+package codersdk
+
+import (
+	"encoding/json"
+	"net/netip"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type ResourceType string
+
+const (
+	ResourceTypeOrganization    ResourceType = "organization"
+	ResourceTypeTemplate        ResourceType = "template"
+	ResourceTypeTemplateVersion ResourceType = "template_version"
+	ResourceTypeUser            ResourceType = "user"
+	ResourceTypeWorkspace       ResourceType = "workspace"
+)
+
+type AuditAction string
+
+const (
+	AuditActionCreate AuditAction = "create"
+	AuditActionWrite  AuditAction = "write"
+	AuditActionDelete AuditAction = "delete"
+)
+
+type AuditDiff map[string]struct {
```

Comment:
the value type of the map should be a named type to get the TypeScript stuff to generate usefully.